### PR TITLE
Rl model example main

### PR DESCRIPTION
**Pull request by cz9779**
This PR fixes [#341 ]. 

### Description of changes
Creates an example for reinforcement learning based data augmentation model.

### Test Conducted
A README that includes code usage and experiment results.

### CODE REVIEW

File: examples/data_augmentation/reinforcement/main.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+        self.optim = tx.core.BertAdam(
+            opt_params, betas=(0.9, 0.999), eps=1e-6, lr=classifier_lr)
+        num_warmup_steps = int(self.num_train_steps
+                               * config_data.warmup_proportion)
+        self.scheduler = torch.optim.lr_scheduler.LambdaLR(
+            self.optim, functools.partial(model_utils.get_lr_multiplier,
+                                     total_steps=self.num_train_steps,
+                                     warmup_steps=num_warmup_steps))
+
+    def pre_train_classifier_epoch(self):
```
Comment (hunterhector) [inline examples/data_augmentation/reinforcement/main.py:161]: I think you need to explain why pre-train is required.

File: examples/data_augmentation/reinforcement/main.py [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+            hparams=config_data.train_hparam, device=device)
+        val_dataset = tx.data.RecordData(
+            hparams=config_data.eval_hparam, device=device)
+        test_dataset = tx.data.RecordData(
+            hparams=config_data.test_hparam, device=device)
+        self.iterator = tx.data.DataIterator(
+            {"train": train_dataset,
+             "eval": val_dataset,
+             "test": test_dataset}
+        )
```
Comment (hunterhector) [inline examples/data_augmentation/reinforcement/main.py:93]: Normally we use `train`, `dev`, `test`. `eval` is normally considered the same as `test`

File: examples/data_augmentation/reinforcement/main.py [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+             'weight_decay': 0.0}]
+        aug_optim = tx.core.BertAdam(
+            optimizer_grouped_parameters, betas=(0.9, 0.999),
+            eps=1e-6, lr=aug_lr)
+        # Builds data augmentation wrapper
+        self.aug_wrapper = MetaAugmentationWrapper(
+            aug_model, aug_optim, input_mask_ids, device, args.num_aug)
+
+    def _init_classifier(self):
+        # Builds downstream BERT classifier
```
Comment (hunterhector) [inline examples/data_augmentation/reinforcement/main.py:126]: Downstream BERT would sound a bit confusing. Probably call it BERT for classification task?

File: examples/data_augmentation/reinforcement/README.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+This model builds upon the connection of supervised learning and reinforcement learning (RL), and adapts reward learning algorithm from RL for joint data augmentation learning and model training.  
+For details, please refer to the [Learning Data Manipulation for Augmentation and Weighting](https://arxiv.org/pdf/1910.12795.pdf) and the original [github](https://github.com/tanyuqian/learning-data-manipulation/).
+
+This example demonstrates the usage of `forte/models/da_rl/MetaAugmentationWrapper`, that wraps a BERT Masked Language Model data augmentation model to perform this RL adaptive learning with a BERT-based text classifier downstream model.
+
+
+### Examplge Usage
+
+See `RLAugmentClassifierTrainer.train_epoch()` in `main.py` for details.
+
```
Comment (hunterhector) [inline examples/data_augmentation/reinforcement/README.md:12]: Can you briefly explain what happens here? Like, why we are wrapping, why we compute validation loss and real loss.

**Review by hunterhector — COMMENTED** [review]
A few comments.

**Review by hunterhector — APPROVED** [review]
lgtm
